controllers/storageoscluster/metrics-exporter_operand.go

```diff
@@ -0,0 +1,195 @@
+package storageoscluster
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/ondat/operator-toolkit/declarative"
+	"github.com/ondat/operator-toolkit/declarative/kubectl"
+	"github.com/ondat/operator-toolkit/declarative/kustomize"
+	eventv1 "github.com/ondat/operator-toolkit/event/v1"
+	"github.com/ondat/operator-toolkit/operator/v1/operand"
+	storageoscomv1 "github.com/storageos/operator/api/v1"
+	"github.com/storageos/operator/internal/image"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/kustomize/api/filesys"
+	kustomizetypes "sigs.k8s.io/kustomize/api/types"
+)
+
+const (
+	// kMetricsExporterImageName is the image we target with kustomize
+	// transformation on getMetricsExporterBuilder(). Must match the one
+	// on the manifest
+	kMetricsExporterImageName  = "storageos/metrics-exporter:v0.1.1"
+	metricsExporterImageEnvVar = "RELATED_IMAGE_METRICS_EXPORTER"
+)
+
+type MetricsExporterOperand struct {
+	name            string
+	client          client.Client
+	requires        []string
+	requeueStrategy operand.RequeueStrategy
+	fs              filesys.FileSystem
+	kubectlClient   kubectl.KubectlClient
+
+	prometheusCRDsInstalled bool
+	prometheusCRDsLock      chan struct{}
+}
+
+var _ operand.Operand = &MetricsExporterOperand{}
+
+func (c *MetricsExporterOperand) Name() string                                           { return c.name }
+func (c *MetricsExporterOperand) Requires() []string                                     { return c.requires }
+func (c *MetricsExporterOperand) RequeueStrategy() operand.RequeueStrategy               { return c.requeueStrategy }
+func (c *MetricsExporterOperand) PostReady(ctx context.Context, obj client.Object) error { return nil }
+
+// dependenciesFound dictates wether the dependencies are present
+// and thus we can apply the metrics-exporter resources
+// currently we depend on Prometheus CRDs
+func (c *MetricsExporterOperand) dependenciesFound() bool {
+	return c.prometheusCRDsInstalled
+}
+
+func (c *MetricsExporterOperand) updateDependenciesState(currentState bool) {
+	c.prometheusCRDsLock <- struct{}{}
+	c.prometheusCRDsInstalled = currentState
+	<-c.prometheusCRDsLock
+}
+
+func (c *MetricsExporterOperand) ReadyCheck(ctx context.Context, obj client.Object) (bool, error) {
+	ctx, span, _, log := instrumentation.Start(ctx, "MetricsExporterOperand.ReadyCheck")
+	defer span.End()
+
+	ctx, cancel := context.WithTimeout(ctx, time.Minute)
+	defer cancel()
+
+	if !c.dependenciesFound() {
+		return false, nil
```

Comment:
Should we log if the dependencies are not found?

Reply:
I don't think we need a log msg here, k8s will have said something about the "metrics-exporter not ready yet" already